src/lib/utils/options/declaration.ts

```diff
@@ -503,6 +526,12 @@ const converters: {
         option.validate?.(value);
         return value;
     },
+    [ParameterType.Object](value, option, _configPath, oldValue) {
+        option.validate?.(value);
```

Comment:
It may be worth writing a default validation to check that the value is indeed as expected or error?

There appears to be no type checking on any of the declaration type values through the abstraction layers, so the above question applies to all of them.

Reply:
That's a fair point, most options today silently perform a conversion... I'm not sure it matters too much since it only really affects people who write a js config file without any type checking...

Reply:
I was thinking more specifically of a userland scenario where a plugin is created:
```ts
app.options.addDeclaration({
    name: 'whatever',
    type: ParameterType.Object,
    defaultValue:{...}
})
```
I can now call;  
`options.setValue('whatever', [..])`,  
`options.setValue('whatever', {..})`,  
`options.setValue('whatever', 'string')`,  
`options.setValue('whatever', true)`  
and as far as type checking and the compiler is concerned, this is all OK.

This may even be OK at runtime until some esoteric error emerges due to data corruption which will be difficult to debug. 

Also, more specific to `Object` only; according to type checking, `defaultValue:[...]`, an array, is OK in the declaration and will safely compile. This will however cause strange behavior for the [object merging](https://github.com/TypeStrong/typedoc/blob/c265735d77c8ce5432f78b1a3a4d77a0145c1d4a/src/lib/utils/options/declaration.ts#L532), which will not fail but will break data structure and cause unexpected errors elsewhere. This is why I suggested default validators, for here something like:
```ts
//for data and maybe oldData do
if(Array.isArray(data) || typeof data !=='object`) throw new Error('data is not an Object')
```

The root of the problem for `oldData` is that the [type checking on overloads](https://github.com/TypeStrong/typedoc/blob/c265735d77c8ce5432f78b1a3a4d77a0145c1d4a/src/lib/utils/options/declaration.ts#L181-L200) allows `Mixed` (and it's clone `Object`) to be an Array Object or an Object Object (curse you javascript!!!!!). I have stared at this type checking code  for hours before conceding that I just cant figure how it works...

This is not really a userland problem with `Mixed`, but is for `Object` 

This is also not a critical issue. More of a pedantic twitch...
Users who declare one datatype and then set another over it probably deserve what they get. Just hope it is not me on a hangover day!